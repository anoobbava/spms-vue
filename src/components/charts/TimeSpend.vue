<template>
  <div>
    <div v-if="ticketActivityLogs.length > 0">
      <h2>Time Spend Chart</h2>
      <line-chart :data="this.dataValues"></line-chart>
    </div>

    <div v-else>
      <h2 text-xs-center>You are not spending on any projects</h2>
      <line-chart :data="this.sampleDataValues"></line-chart>
    </div>
  </div>
</template>

<script>
import { mapGetters } from 'vuex'
export default {
  data () {
    return {
      sampleDataValues: [['2017-01-01', 1], ['2017-01-29', 1], ['2017-02-01', 2],
        ['2017-02-27', 3], ['2017-03-01', 5], ['2017-03-29', 8],
        ['2017-04-01', 13], ['2017-04-30', 21], ['2017-05-01', 34], ['2017-05-30', 55]]

    }
  },

  computed: {
    ...mapGetters(['ticketActivityLogs']),
    dataValues () {
      const a = this.createChartData()
      return a
    }
  },

  methods: {
    createChartData () {
      const dataArray = []
      for (let i = 0; i < this.ticketActivityLogs.length; i += 1) {
        const localArray = []
        localArray.push(this.ticketActivityLogs[i].attributes.log_date)
        localArray.push(this.ticketActivityLogs[i].attributes.log_time)
        dataArray.push(localArray)
      }
      // debugger
      return dataArray
    }
  }
}
</script>
