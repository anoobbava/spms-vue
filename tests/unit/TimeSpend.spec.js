import { shallowMount, createLocalVue } from '@vue/test-utils'
import VueRouter from 'vue-router'
import TimeSpend from '@/components/charts/TimeSpend.vue'
import Vuex from 'vuex'

describe('TimeSpend', () => {
  const localVue = createLocalVue()
  localVue.use(VueRouter)
  localVue.use(Vuex)
  const router = new VueRouter()
  const store = new Vuex.Store({
    state: {
      ticketActivityLogs: [
        {
          attributes: {
            activity: 'Test Activity1',
            log_date: '2019-01-01',
            ticket_title: 'Test Ticket',
            ticket_id: 'test1',
            status: 'Testing'
          }
        }
      ]
    },

    getters: {
      ticketActivityLogs: (state) => state.ticketActivityLogs
    }
  })
  const wrapper = shallowMount(TimeSpend, { localVue, router, store })

  it('is a component', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  it('renders correctly', () => {
    expect(wrapper.$el).toMatchSnapshot()
  })
})
