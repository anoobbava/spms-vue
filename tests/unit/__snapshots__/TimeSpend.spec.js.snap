// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeSpend renders correctly 1`] = `undefined`;
